addons/crm_survey/tests/test_crm_survey.py

```diff
@@ -0,0 +1,257 @@
+                None, 'How old are you?', 'simple_choice',
+                sequence=1,
+                constr_mandatory=True, constr_error_msg='Please select an answer', survey_id=survey.id,
+                labels=[
+                    {'value': '18-30', 'is_create_lead': False},
+                    {'value': '30-50', 'is_create_lead': False},
+                    {'value': '50+', 'is_create_lead': False},
+                ])
+
+            q02 = self._add_question(
+                None, 'What programming languages do you use on a daily basis?', 'multiple_choice',
+                sequence=2,
+                constr_mandatory=True, constr_error_msg='Please select an answer', survey_id=survey.id,
+                labels=[
+                    {'value': 'Assembly', 'is_create_lead': True},
+                    {'value': 'Java', 'is_create_lead': False},
+                    {'value': 'C', 'is_create_lead': False},
+                    {'value': 'Python', 'is_create_lead': False},
+                ])
+
+            q03 = self._add_question(
+                None, 'How many years of experience do you have in this position?', 'simple_choice',
+                sequence=3,
+                constr_mandatory=True, constr_error_msg='Please select an answer', survey_id=survey.id,
+                labels=[
+                    {'value': '0-1 year', 'is_create_lead': True},  # Newbie power
+                    {'value': '2-5 years', 'is_create_lead': False},
+                    {'value': '6+ years', 'is_create_lead': False},
+                ])
+
+            q04 = self._add_question(
+                None, 'Please skip this one', 'simple_choice',
+                sequence=4,
+                constr_mandatory=False, survey_id=survey.id,
+                labels=[
+                    {'value': 'No.', 'is_create_lead': False},
+                    {'value': 'NO', 'is_create_lead': False},
+                    {'value': 'OK, btw', 'is_create_lead': False},
+                ])
+
+            q05 = self._add_question(
+                None, 'Please add or verify your email address :', 'char_box',
+                sequence=5,
+                validation_email=True,
+                constr_mandatory=True, constr_error_msg='Please select an answer', survey_id=survey.id,
+                )
+
+        self.assertFalse(q01.is_lead_generating)
+        self.assertTrue(q02.is_lead_generating)
+        self.assertTrue(q03.is_lead_generating)
+        self.assertFalse(q04.is_lead_generating)
+        self.assertFalse(q05.is_lead_generating)
+
+        return survey
+
+    def test_connected_account_access_with_lead_generation_answer(self):
+        # Step 1 : Connected access + lead generation
+        survey = self._create_lead_qualification_survey(survey_type="custom")
+
+        # Account connection
+        login_password = 'survey_user'
+        user = self.env['res.users'].search([('login', '=', login_password)])
+
+        with self.with_user(login_password):
```

Comment:
I am not sure that you're doing what you want to do ^^

Did you want to authenticate as a given user?

then check self.authenticate(...) ((None, None) argument corresponds to the public user)

I know that survey is a bit different, so maybe it works in an unusual way, but this feels weird. We usually use it as a way to read / write records + check access rights. Could work, just strange from my point of view

(Side note: by default, tests are run as a superadmin, having all rights)